ecommerce/extensions/api_client/get_smarter.py

```diff
@@ -0,0 +1,106 @@
+
+import datetime
+import logging
+
+import pytz
+import requests
+from django.conf import settings
+from edx_django_utils.cache import TieredCache
+from oauthlib.oauth2 import BackendApplicationClient
+from requests_oauthlib import OAuth2Session
+
+logger = logging.getLogger(__name__)
+
+
+class GetSmarterEnterpriseApiClient:
+    """
+    Client to interface with the GetSmarter Enterprise API Gateway(GEAG).
+    """
+
+    def __init__(self):
+        required_settings = [
+            'GET_SMARTER_OAUTH2_KEY',
+            'GET_SMARTER_OAUTH2_SECRET',
+            'GET_SMARTER_OAUTH2_PROVIDER_URL',
+            'GET_SMARTER_API_URL'
+        ]
+
+        for setting in required_settings:
+            if not getattr(settings, setting, None):
+                logger.error('Failed to initialize GetSmarterEnterpriseApiClient, missing %s.', setting)
+                raise ValueError('Missing {setting}.')
+
+        self.oauth_client_id = settings.GET_SMARTER_OAUTH2_KEY
+        self.oauth_client_secret = settings.GET_SMARTER_OAUTH2_SECRET
+        self.oauth_provider_url = settings.GET_SMARTER_OAUTH2_PROVIDER_URL
+        self.api_url = settings.GET_SMARTER_API_URL
+        self.session = requests.Session()
+
+    @property
+    def oauth_token_url(self):
+        return self.oauth_provider_url + '/oauth2/token'
+
+    @property
+    def access_token_cache_key(self):
+        return 'get_smart_enterprise_gateway.access_token.{}'.format(self.oauth_client_id)
+
+    def _get_cached_access_token(self):
+        """
+        Return the cached access token if it is not expired.
+        """
+        cached_response = TieredCache.get_cached_response(self.access_token_cache_key)
+
+        if cached_response.is_found:
+            cached_value = cached_response.value
+            expires_at = cached_value['expires_at']
+            if datetime.datetime.now(pytz.utc).timestamp() < expires_at:
```

Comment:
bah, I wish the request cache supported a timeout, too (as opposed to just the Django cache portion of `TieredCache`), so that we wouldn't have to check this expiration.